src/modals/EditDocumentModal.js

```diff
@@ -11,63 +11,66 @@ const EditDocumentModal = ({
   updateFile,
   updateVersion,
   onCloseDialog,
+  study,
   user,
 }) => {
-  const [fileTypeInput, setFileType] = useState(fileNode.fileType);
-  const [fileNameInput, setFileName] = useState(fileNode.name);
-  const [fileDescriptionInput, setFileDescription] = useState(
-    fileNode.description,
-  );
+  const formEl = useRef(null);
 
   const latestVersion = fileSortedVersions(fileNode)[0].node;
-  const [versionStatusInput, setVersionStatus] = useState(latestVersion.state);
+
   const isAdmin = !user.loading
     ? user.myProfile.roles.includes('ADMIN')
     : false;
 
-  const onSubmit = e => {
-    e.preventDefault();
-    const name = fileNameInput;
-    const description = fileDescriptionInput;
-    const fileType = fileTypeInput;
-    updateFile({variables: {kfId: fileNode.kfId, name, description, fileType}})
-      .then(() => onCloseDialog())
-      .catch(err => console.log(err));
-    updateVersion({
-      variables: {
-        versionId: latestVersion.kfId,
-        description: latestVersion.description,
-        state: versionStatusInput,
-      },
-    })
-      .then(() => onCloseDialog())
-      .catch(err => console.log(err));
+  const handleSubmit = async (name, fileType, description, versionStatus) => {
+    try {
+      await updateFile({
```

Comment:
Do you intend to do these synchronously, or can you use `Promise.all()`?

Reply:
this was a refactor of the existing function to attempt to catch all the errors in one place instead of individually. Wasn't sure if the order of operations mattered here when saving